src/lib/output/ResultsProcessorFactory.ts

```diff
@@ -0,0 +1,35 @@
+import {Display} from "../Display";
+import {JsonReturnValueHolder} from "./JsonReturnValueHolder";
+import {RunOutputOptions, RunResultsProcessor} from "./RunResultsProcessor";
+import {CompositeResultsProcessor, ResultsProcessor} from "./ResultsProcessor";
+import {ENV_VAR_NAMES} from "../../Constants";
+import {inferFormatFromInternalOutfile, OutputFormat} from "./OutputFormat";
+import {OutfileResultsProcessor} from "./OutfileResultsProcessor";
+
+/**
+ * Interface for creating a ResultsProcessor
+ */
+export interface ResultsProcessorFactory {
+	createResultsProcessor(display: Display, runOutputOptions: RunOutputOptions,
+							jsonReturnValueHolder: JsonReturnValueHolder): ResultsProcessor
+}
+
+/**
+ * Runtime implementation of the ResultsProcessorFactory interface
+ */
+export class ResultsProcessorFactoryImpl implements ResultsProcessorFactory {
+	public createResultsProcessor(display: Display, runOutputOptions: RunOutputOptions,
+									jsonReturnValueHolder: JsonReturnValueHolder): ResultsProcessor {
+		const resultsProcessors: ResultsProcessor[] = [new RunResultsProcessor(display, runOutputOptions, jsonReturnValueHolder)];
+		this.addProcessorForInternalOutfileIfNeeded(resultsProcessors, runOutputOptions.verboseViolations);
+		return new CompositeResultsProcessor(resultsProcessors);
```

Comment:
In the future we can put a lot of the business logic of which result processors should be added based on the run options (or inputs) here. Not much going on yet... but I would love to revisit this at some point if this turns into the pattern for v4 cli.